File: LeetCode_Algorithm/1431(Kids With the Greatest Number of Candies)/1431.js
var kidsWithCandies = function(candies, extraCandies) {
  // sol1 63ms 57% 42.3MB 56%
  let maxCandy = Math.max(...candies)  
  for (let i=0; i < candies.length; i++) {
    if (candies[i] + extraCandies >= maxCandy) {
      candies[i] = true
    } else {
      candies[i] = false
    }
  }
  return candies
};

File: LeetCode_Algorithm/1431(Kids With the Greatest Number of Candies)/1431.py
'''
LeetCode 1431 : Kids With the Greatest Number of Candies

# sol1 38ms 75% 13.8MB 95%

'''
class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        max_candies = max(candies)
        for i in range(len(candies)):
            if candies[i] + extraCandies >= max_candies:
                candies[i] = True
            else:
                candies[i] = False
        return candies